Fix QCefView object injection and makefile for building outside of the source directory

Klarso-IvoBurkart opened the pull request.

Klarso-IvoBurkart committed:
Fix test makefile for building outside of the source directory

Klarso-IvoBurkart committed:
Add QCefView::runJavaScript() to easily inject JS into the browser

Klarso-IvoBurkart committed:
Fix: Create render delegates, so the QCefView object is actually injected into JS engine

This makes the test app work again (mostly).

Klarso-IvoBurkart committed:
Merge remote-tracking branch 'upstream/master'

tishion commented:
hi @Klarso-IvoBurkart , thank you for the commits, though I don't know whether the JS injection is widely used by people I am megring this PR to this repo. 

BTW, I have reformed this project by imporiving the hierarchy and layer splitting, I just haven't make it public but i will invite you to the private repo now.

Klarso-IvoBurkart committed:
Add source and line to consoleMessage event

Klarso-IvoBurkart committed:
Use browser identifiers rather than pointers for DefaultSchemeHandler

This fixes the URL scheme event in the test

Klarso-IvoBurkart committed:
Show popups for console messages in test app

Klarso-IvoBurkart committed:
Add QCefSchemeHandler to make it easy to provide your own data via a custom url scheme.

See test app for usage.

Klarso-IvoBurkart committed:
Expose onAddressChange and onTitleChange events

Klarso-IvoBurkart commented:
> hi @Klarso-IvoBurkart , thank you for the commits, though I don't know whether the JS injection is widely used by people I am megring this PR to this repo.
> 
> BTW, I have reformed this project by imporiving the hierarchy and layer splitting, I just haven't make it public but i will invite you to the private repo now.

Hi @tishion ,

thanks for the invite to the repo. I’ll have a look next week.

It actually made some more changes and pushed them a few minutes ago. Hopefully the can be integrated into the new structure easily.

Maybe you can have a look before I create a pull request :-)

Have a nice weekend,
Ivo

Klarso-IvoBurkart committed:
Expose find text API

Klarso-IvoBurkart committed:
Expose zoom level API

Klarso-IvoBurkart committed:
Reemit key events to the browser view

Klarso-IvoBurkart committed:
For more stability, check that the first loadend event has arrived, before
allowing access to the browser

Klarso-IvoBurkart committed:
Default implementation for onAddressChange and onTitleChange

tishion commented:
Hi @Klarso-IvoBurkart, thank you for the commits and sorry for late reply (cos' i have just moved across border).

I just reviewed your great work and added one small comment on the introducing of the semaphore. can you share some thoughts on this?

Klarso-IvoBurkart commented:
> Hi @Klarso-IvoBurkart, thank you for the commits and sorry for late reply (cos' i have just moved across border).
> 
> I just reviewed your great work and added one small comment on the introducing of the semaphore. can you share some thoughts on this?

Hi @tishion,

I can't seem to find your comment. Since we always construct the browser with an address (or about:blank) we will get a LoadEnd event if everything is set up. If we try to call any API before that (i.e. directly after constructing the view) crashes will happen or the calls will be ignored. Thats why the exposed API-Functions wait for the first LoadEnd to arrive to make sure the browser is fully initialized.

tishion commented:
Give me some time and I am looking into it.

tishion commented:
@Klarso-IvoBurkart ， really sorry for long pending, I haven't got full Internet access from my new place. i will handle the PR this weekends.  Thanks!

tishion commented:
I will not accept this PR until you addressed all the comments. Thanks.
